python/cuml/preprocessing/text/stem/porter_stemmer.py

```diff
@@ -173,7 +174,7 @@ def _step1a(self, word_str_ser, can_replace_mask=None):
             )
 
             # update can replace mask
-            can_replace_mask = can_replace_mask & cudf.logical_not(
+            can_replace_mask = can_replace_mask & np.logical_not(
                 condition_mask
             )
```

Comment:
I think this PR can use `~condition_mask` instead of `np.logical_not(condition_mask)`. It matches better to the use of the bitwise and operator `&` in the same expression. (Kind of like writing `a + b` instead of `np.add(a, b)`.) I think this can also use an in-place bitwise and.
```suggestion
            can_replace_mask &= ~condition_mask
```

Reply:
Also could help avoid some uncertainty for future readers about whether the code is operating on a GPU or CPU backed object

Reply:
Done.